File: 01-Week-1/01-Problem-Set-1/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int change_in_cents;
    do
    {
        change_in_cents = get_int("Change owed: ");
    } while (change_in_cents < 0);

    const int quarters = 25;
    const int dimes = 10;
    const int nickels = 5;
    const int pennies = 1;

    int quarters_count = 0;
    int dimes_count = 0;
    int nickels_count = 0;
    int pennies_count = 0;

    int total_coins_count = 0;

    int sum_quarters = 0;
    int sum_dimes = 0;
    int sum_nickels = 0;
    int sum_pennies = 0;

    if (change_in_cents / quarters != 0)
    {
        // Calculate how many quarters you should give customer
        quarters_count = change_in_cents / quarters;
        total_coins_count += quarters_count;
        sum_quarters = quarters_count * quarters;
        // Subtract the value of those quarters from cents
        change_in_cents = change_in_cents - sum_quarters;
    }

    if (change_in_cents / dimes != 0)
    {
        // Calculate how many dimes you should give customer
        dimes_count = change_in_cents / dimes;
        total_coins_count += dimes_count;
        sum_dimes = dimes_count * dimes;
        // Subtract the value of those dimes from remaining cents
        change_in_cents = change_in_cents - sum_dimes;
    }

    if (change_in_cents / nickels != 0)
    {
        // Calculate how many nickels you should give customer
        nickels_count = change_in_cents / nickels;
        total_coins_count += nickels_count;
        sum_nickels = nickels_count * nickels;
        // Subtract the value of those nickels from remaining cents
        change_in_cents = change_in_cents - sum_nickels;
    }

    if (change_in_cents / pennies != 0)
    {
        // Calculate how many pennies you should give customer
        pennies_count = change_in_cents / pennies;
        total_coins_count += pennies_count;
        sum_pennies = pennies_count * pennies;
        // Subtract the value of those pennies from remaining cents
        change_in_cents = change_in_cents - sum_pennies;
    }

    // Print the sum of quarters, dimes, nickels, and pennies used
    printf("%i\n", total_coins_count);
}

File: 06-Week-6/01-Problem-Set-6/sentimental-cash/cash.py
from cs50 import get_float

quarters = 0.25 * 100
dimes = 0.10 * 100
nickels = 0.05 * 100
pennies = 0.01 * 100

quarters_count = 0
dimes_count = 0
nickels_count = 0
pennies_count = 0

money = 0
total_coins_count = 0

# Prompt the user for change owed, in cents.
while True:
    user_input = get_float("Change: ")
    if user_input > 0:
        money = round(user_input * 100)
        break

# Calculate how many quarters you should give customer. Subtract the value of those quarters from cents.
if money / quarters != 0:
    quarters_count += int(money / quarters)
    total_coins_count += quarters_count
    money -= quarters * quarters_count

# Calculate how many dimes you should give customer. Subtract the value of those dimes from remaining cents.
if money / dimes != 0:
    dimes_count += int(money / dimes)
    total_coins_count += dimes_count
    money -= dimes * dimes_count

# Calculate how many nickels you should give customer. Subtract the value of those nickels from remaining cents.
if money / nickels != 0:
    nickels_count += int(money / nickels)
    total_coins_count += nickels_count
    money -= nickels * nickels_count

# Calculate how many pennies you should give customer. Subtract the value of those pennies from remaining cents.
if money / pennies != 0:
    pennies_count += int(money / pennies)
    total_coins_count += pennies_count
    money -= pennies * pennies_count

# Print that sum
print(total_coins_count)